src/transformers/generation/utils.py

```diff
@@ -350,47 +350,69 @@ def prepare_inputs_for_generation(
         attention_mask: Optional[torch.LongTensor] = None,
         inputs_embeds: Optional[torch.FloatTensor] = None,
         cache_position: Optional[torch.LongTensor] = None,
-        position_ids: Optional[torch.LongTensor] = None,
```

Comment:
Not all models expect this one. We now inspect the signature to determine whether we need to generate them on the fly